File: server/app/api/v1/classrooms.py
# classrooms.py — 空闲教室 API (P4.6)
"""
GET /api/v1/classrooms — 查询空闲教室
"""
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession
from app.database.session import get_db
from app.database.repositories.repos import ClassroomRepo
from app.auth.dependencies import get_current_user
from app.database.models import User

router = APIRouter()


@router.get("")
async def get_classrooms(
    week_day: int = Query(None, ge=1, le=7, description="星期 (1-7)"),
    building: str = Query(None, description="教学楼名 (可选模糊)"),
    user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    """获取空闲教室列表"""
    repo = ClassroomRepo(db, user.id)
    if week_day is not None:
        classrooms = await repo.get_by_weekday(week_day, building)
    else:
        classrooms = await repo.get_all()

    if building and week_day is None:
        # 非索引筛选：内存过滤
        classrooms = [c for c in classrooms
                      if building in (c.building or "")]

    return {
        "total": len(classrooms),
        "data": [
            {
                "campus": c.campus,
                "building": c.building,
                "room_no": c.room_no,
                "capacity": c.capacity,
                "week_day": c.week_day,
                "start_node": c.start_node,
                "end_node": c.end_node,
                "status": c.status,
            }
            for c in classrooms
        ],
    }

File: app/src/api/classrooms.ts
// ============================================================
// 空闲教室 API — GET /api/v1/classrooms
// ============================================================
import apiClient from './client';

export interface ClassroomItem {
  campus: string;
  building: string;
  room_no: string;
  capacity: number;
  week_day: number;
  start_node: number;
  end_node: number;
  status: string;
}

export interface ClassroomListResponse {
  total: number;
  data: ClassroomItem[];
}

export const classroomsApi = {
  /** 查询空闲教室 */
  getClassrooms(weekDay?: number, building?: string): Promise<ClassroomListResponse> {
    const params: Record<string, string | number> = {};
    if (weekDay !== undefined) params.week_day = weekDay;
    if (building) params.building = building;
    return apiClient.get('/classrooms', { params }).then((r) => r.data);
  },
};
